holoviews/operation/downsample.py

```diff
@@ -256,7 +256,7 @@ def _process(self, element, key=None, shared_data=None):
 
         if len(element) <= self.p.width:
             return element
-        xs, ys = (element.dimension_values(i) for i in range(2))
+        xs, ys = (np.ascontiguousarray(element.dimension_values(i)) for i in range(2))
```

Comment:
this could potentially be a heavy computation, if this is only needed for the `tsdownsample` algorithms I would move this up into those functions.